src/olympia/devhub/decorators.py

```diff
@@ -32,11 +31,18 @@ def wrapper(request, addon, *args, **kw):
             def fun():
                 return f(request, addon_id=addon.id, addon=addon, *args, **kw)
 
+            read_permissions = [
+                amo.permissions.REVIEWER_TOOLS_VIEW,
+                amo.permissions.REVIEWER_TOOLS_UNLISTED_VIEW,
+            ]
+
             if request.method in ('HEAD', 'GET'):
                 # Allow reviewers for read operations.
                 if allow_reviewers_for_read and (
                     acl.is_reviewer(request, addon)
-                    or acl.action_allowed(request, permissions.REVIEWER_TOOLS_VIEW)
+                    or any(
+                        acl.action_allowed(request, perm) for perm in read_permissions
+                    )
```

Comment:
This would not be ideal. It would allow read access in devhub to users with the unlisted view permission and not the other one, regardless of what channels the add-on has versions in... This might be ok, but it can be a bit surprising...

Reply:
Do you know why we have `REVIEWER_TOOLS_VIEW` at all here? What is that needed for?

Reply:
Do you know why we have `REVIEWER_TOOLS_VIEW` at all here? What is that needed for?

Reply:
Ah, this is just used to access the file validation page or json, which is part of the devhub. This is controlled by `allow_reviewers_for_read`

Reply:
Ah, yeah. We were already inconsistent (previously allowing `REVIEWER_TOOLS_VIEW` users access to unlisted) so I suppose this is not worse with your changes, but I'm concerned about the potential for misusing this in the future...

Can you update the docstring? Instead of referencing `allow_reviewers` which is no longer a thing, it should mention what `allow_reviewers_for_read` does, updated with your changes. Hopefully that should be enough.

Reply:
I updated the decorator as per the Matrix chat in eeec85729e.